back-end/src/controllers/featureController.ts

```diff
@@ -285,4 +286,57 @@ async function deleteEvent(req: Request, res: Response) {
   }
 }
 
-export default { editProfile, createEvent, getEvent, updateEvent, deleteEvent };
+async function createForum(req: Request, res: Response) {
+  try {
+    let decoded = (<any>req).decoded;
+    let {
+      forum_name,
+      category,
+      description,
+      image,
+    } = req.body;
+
+    let user = await userModel.getUserData(decoded);
+
+    let id_user = user.data.id;
+    console.log("ID User : " + id_user);
+
+    if (user.data.user_role !== 'Admin') {
+      res.status(SERVER_OK).json({
+        success: false,
+        data: {},
+        message: 'Only admin can create a forum',
+      });
+      return;
+    }
+
+    if (!forum_name || !category || !description) {
+      res.status(SERVER_OK).json({
+        success: false,
+        data: {},
+        message: 'Please fill all required fields',
+      });
+      return;
+    }
+
+    image = image ? image : null;
+
+    let result = await forumsModel.newForum({
+      id_user,
+      forum_name,
+      category,
+      description,
+      image,
+    });
+
+    if (result.success) {
+      res.status(SERVER_OK).json(result);
+    } else {
+      res.status(SERVER_BAD_REQUEST).json(result);
+    }
+  } catch (e) {
+    res.status(SERVER_BAD_REQUEST).json(String(e));
+  }
+}
+
+export default { editProfile, createEvent, getEvent, createForum, updateEvent, deleteEvent };
```

Comment:
Place the createForum at the end of the export. Ex : export default {editProfile, createEvent, getEvent, updateEvent, deleteEvent, createForum)

Reply:
This is for a better readability